internal/validate/vsa/vsa.go

```diff
@@ -220,3 +223,164 @@ func NoopUploader(ctx context.Context, att oci.Signature, location string) (stri
 	log.Infof("Upload type is 'none'; skipping upload for %s", location)
 	return "", nil
 }
+
+// VSALookupResult represents the result of looking up an existing VSA
+type VSALookupResult struct {
+	Found     bool
+	Expired   bool
+	VSA       *Predicate
+	Timestamp time.Time
+}
+
+// VSAChecker handles checking for existing VSAs in Rekor
+type VSAChecker struct {
+	RekorURL string
+	Timeout  time.Duration
+}
+
+// NewVSAChecker creates a new VSA checker
+func NewVSAChecker(rekorURL string, timeout time.Duration) *VSAChecker {
```

Comment:
This is also coupled with Rekor. It could accept a `retriever` interface.

Reply:
Thanks, this now accepts VSARetriever